dgl_sparse/src/sparse_format.cc

```diff
@@ -113,7 +113,8 @@ std::shared_ptr<CSR> DiagToCSR(
     const c10::TensorOptions& indices_options) {
```

Comment:
not related to this PR, why do we need to pass indices_options in instead of create it here?

Reply:
why other conversion does not need the option. e.g. COOToCSC

Reply:
This is used to decide the dtype and devices of the format to convert. The Diag sparse format only have `num_rows` and `num_cols` without those information.